src/sentry/models/integrations/integration.py

```diff
@@ -60,6 +58,28 @@ def get_provider(self):
 
         return integrations.get(self.provider)
 
+    def delete(self, *args, **kwds):
+        with transaction.atomic(), test_psql_role_override("postgres"):
+            for outbox in Integration.outboxes_for_update(self.id):
```

Comment:
Using an outbox almost certainly should following within the same atomic transaction as some other write that it is replicating.  Not using transaction.atomic runs the risk of integrity issues (partial writes).